conans/client/cmd/export.py

```diff
@@ -504,11 +498,27 @@ def export_recipe(conanfile, origin_folder, destination_folder):
         copier(pattern, links=True, excludes=excluded_exports)
     copier.report(package_output)
 
-    if hasattr(conanfile, "export"):
-        if not callable(conanfile.export):
-            raise ConanException("conanfile 'export' must be a method")
+    _run_method(conanfile, "export", origin_folder, destination_folder, output)
+
+
+def _run_method(conanfile, method, origin_folder, destination_folder, output):
+    export_method = getattr(conanfile, method, None)
+    if export_method:
+        if not callable(export_method):
+            raise ConanException("conanfile '%s' must be a method" % method)
         copier = FileCopier([origin_folder], destination_folder)
         conanfile.copy = copier
-        conanfile.export()
-        export_method_output = ScopedOutput("%s export() method" % output.scope, output)
+        folder_name = "%s_folder" % method
+        setattr(conanfile, folder_name, destination_folder)
+        default_options = conanfile.default_options
+        try:
+            # TODO: Poor man attribute control access. Convert to nice decorator
+            conanfile.default_options = None
+            with chdir(origin_folder):
+                with conanfile_exception_formatter(str(conanfile), method):
+                    export_method()
```

Comment:
We usually have a line like the following before running the methods

```python
output.highlight("Calling xxxxxxx()")
```

Reply:
i believe it just a bit higher up https://github.com/conan-io/conan/pull/6945/files#diff-36cec48ed0992fd835c1bf1c633fc030R509